core/src/Kokkos_MemoryPool.hpp

```diff
@@ -782,16 +782,22 @@ class MemoryPool {
   KOKKOS_INLINE_FUNCTION
   int number_of_superblocks() const noexcept { return m_sb_count; }
 
+  template <class MemorySpace>
+  static KOKKOS_FUNCTION constexpr bool may_access() {
+    KOKKOS_IF_ON_HOST((return SpaceAccessibility<DefaultHostExecutionSpace,
+                                                 MemorySpace>::accessible;))
+    KOKKOS_IF_ON_DEVICE((return SpaceAccessibility<DefaultExecutionSpace,
+                                                   MemorySpace>::accessible;))
+  }
+
   KOKKOS_INLINE_FUNCTION
   void superblock_state(int sb_id, int &block_size, int &block_count_capacity,
                         int &block_count_used) const noexcept {
     block_size           = 0;
     block_count_capacity = 0;
     block_count_used     = 0;
 
-    if (Kokkos::Impl::MemorySpaceAccess<
-            Kokkos::Impl::ActiveExecutionMemorySpace,
-            base_memory_space>::accessible) {
+    if (may_access<base_memory_space>()) {
```

Comment:
Hmm... not sure if I'm in favor of this change.

Reply:
What alternative would you like to suggest?

Reply:
I didn't like defining a separate function for that. I would prefer inlinng it. Something like
```C++
bool may_access = [] () {
  KOKKOS_IF_ON_HOST((return SpaceAccessibility<DefaultHostExecutionSpace, base_memory_space>::accessible;))
  KOKKOS_IF_ON_DEVICE((return SpaceAccessibility<DefaultExecutionSpace, base_memory_space>::accessible;))
}();
```

Reply:
Done in d4a7d45cf